### ISSUE TITLE
StackOverflowError when processing large batches of data

### ISSUE THREAD

**Issue by dmekhanikov** [espertechinc/esper#29]
When running the following EPL:
```sql
create schema Event(id int);

create schema Incident(id int);

create window IncidentWindow.win:keepall() as Incident;

insert into Incident select * from Event.win:expr_batch(current_count >= 10000);

insert into IncidentWindow select * from Incident;

create dataflow EventsDataFlow
    BeaconSource -> events<Event> {id: 1, iterations: 10000}
    EventBusSink(events) {};
```
having EventsDataFlow started StackOverflowError is thrown:
```
java.lang.StackOverflowError: null
	at com.espertech.esper.epl.named.NamedWindowRootViewInstance.addNewData(NamedWindowRootViewInstance.java:101) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.epl.named.NamedWindowTailViewInstance.update(NamedWindowTailViewInstance.java:62) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.view.ViewSupport.updateChildren(ViewSupport.java:115) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.view.window.KeepAllView.update(KeepAllView.java:106) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.view.ViewSupport.updateChildren(ViewSupport.java:115) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.epl.named.NamedWindowRootViewInstance.update(NamedWindowRootViewInstance.java:126) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.view.ZeroDepthStreamNoIterate.insert(ZeroDepthStreamNoIterate.java:50) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.context.activator.ViewableActivatorFilterProxy$2.matchFound(ViewableActivatorFilterProxy.java:98) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processStatementFilterMultiple(EPRuntimeImpl.java:1198) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processMatches(EPRuntimeImpl.java:1071) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueueUnlatched(EPRuntimeImpl.java:934) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueueFront(EPRuntimeImpl.java:855) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:810) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
	at com.espertech.esper.core.service.EPRuntimeImpl.processThreadWorkQueue(EPRuntimeImpl.java:836) ~[esper-5.3.0.jar:na]
...
```

**Comment by bernhardttom** [comment]
Add a test class to reproduce please.

**Comment by dmekhanikov** [comment]
Here it is:
```java
import com.espertech.esper.client.EPRuntime;
import com.espertech.esper.client.EPServiceProvider;
import com.espertech.esper.client.EPServiceProviderManager;
import com.espertech.esper.client.deploy.DeploymentException;
import com.espertech.esper.client.deploy.ParseException;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class EsperBreaker {
    public static void main(String ... args) throws ParseException, DeploymentException, IOException {
        EPServiceProvider epService = EPServiceProviderManager.getDefaultProvider();
        EPRuntime runtime = epService.getEPRuntime();

        epService.getEPAdministrator().getDeploymentAdmin().parseDeploy(
                "create schema Event(id int);\n" +
                "create schema Incident(id int);\n" +
                "create window IncidentWindow.win:keepall() as Incident;\n" +
                "insert into Incident select * from Event.win:expr_batch(current_count >= 10000);\n" +
                "insert into IncidentWindow select * from Incident;\n");

        Map<String, Integer> event = new HashMap<>();
        event.put("id", 1);
        for (int i = 0; i < 10000; i++) {
            runtime.sendEvent(event, "Event");
        }
    }
}
```

This patch should help:
```diff
--- esper/src/main/java/com/espertech/esper/core/service/EPRuntimeImpl.java	(date 1442325065000)
+++ esper/src/main/java/com/espertech/esper/core/service/EPRuntimeImpl.java	(revision )
@@ -833,7 +833,7 @@
             }
 
             if (!queues.getFrontQueue().isEmpty()) {
-                processThreadWorkQueue();
+                processThreadWorkQueueFront(queues);
             }
         }
     }
```

**Comment by bernhardttom** [comment]
thanks great - merged for 5.4 release
